apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx

```diff
@@ -0,0 +1,125 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import Button from '@cdo/apps/templates/Button';
+import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
+
+export default class CloneLessonDialog extends Component {
+  static propTypes = {
+    lessonId: PropTypes.number,
+    lessonName: PropTypes.string,
+    handleClose: PropTypes.func.isRequired
+  };
+
+  defaultState = {
+    destinationScript: '',
+    saving: false,
+    cloneFailed: false,
+    cloneSucceeded: false,
+    responseData: null
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {...this.defaultState};
+  }
+
+  onCloneClick = () => {
+    this.setState({saving: true});
+    const csrfContainer = document.querySelector('meta[name="csrf-token"]');
+    let success = false;
+
+    return fetch(`/lessons/${this.props.lessonId}/clone`, {
+      method: 'POST',
+      body: JSON.stringify({
+        destinationScriptName: this.state.destinationScript
```

Comment:
I have renaming on the brain and it reminded me that we are trying to move from Script to Unit. Should we try to use Unit here?

Reply:
Sounds good! Is the script to unit change a known move? Wondering if/how I should update the text on this dialog -- maybe something like "Which script/unit do you want to clone this lesson to?"

Reply:
It's part of our Q2 plan - https://codedotorg.atlassian.net/browse/PLAT-791. I don't know how deep of a dive we were planning to try to do to move away from script